testpilot/frontend/static-src/app/views/email-opt-in-dialog-view.js

```diff
@@ -0,0 +1,135 @@
+      </header>
+      <div class="modal-content centered">
+        <img class="email-icon" src="/* TODO: add this image to the images dir */">
+        <p data-l10n-id="emailOptInSuccessMessage">When you get a chance, head over to your inbox to verify your email address. Thank you!</p>
+        <button class="button default" data-hook="continue" data-l10n-id="emailOptInConfirmationClose">On to the experiments...</button>
+      </div>
+    </div>
+  </div>
+  `,
+
+  session: {
+    isFirstPage: { type: 'boolean', default: true },
+    isValidEmail: { type: 'boolean', default: true }
+  },
+
+  bindings: {
+    'isFirstPage': [
+      {
+        type: 'toggle',
+        hook: 'first-page'
+      },
+      {
+        type: 'toggle',
+        hook: 'second-page',
+        invert: true
+      }
+    ],
+    'isValidEmail': {
+      type: 'toggle',
+      hook: 'invalid-email',
+      invert: true
+    }
+  },
+
+  events: {
+    'click button[data-hook=submit-email]': 'submit',
+    'click [data-hook=cancel-modal]': 'skip',
+    'click [data-hook=continue]': 'continue'
+  },
+
+  submit(e) {
+    e.preventDefault();
+
+    // TODO: should we log the number of email validity failures? worth tracking?
+    const email = this.query('input[data-hook=email]').value;
+    if (!this.validate(email)) {
+      console.log('email is invalid.');
+      this.set('isValidEmail', false);
+      return;
+    }
+    // Hide the email validation error message, if it's visible.
+    this.set('isValidEmail', true);
+
+    app.sendToGA('event', {
+      eventCategory: 'HomePage Interactions',
+      eventAction: 'button click',
+      eventLabel: 'Sign me up'
+    });
+
+    app.subscribeToBasket(email, () => {
+      // TODO: review question: basket failures are swallowed by the subscribeToBasket
+      // function. Is it worth it to send a second GA ping to help us monitor the error
+      // rate? If so, should I overload the button click interaction, as done below, or
+      // use some other kind of category / action?
+      app.sendToGA('event', {
+        eventCategory: 'HomePage Interactions',
+        eventAction: 'button click',
+        eventLabel: 'email submitted to basket'
+      });
+    });
+
+    this.set('isFirstPage', false);
+  },
+
+  skip(e) {
+    e.preventDefault();
+    e.stopPropagation();
+
+    app.sendToGA('event', {
+      eventCategory: 'HomePage Interactions',
+      eventAction: 'button click',
+      // review TODO: The label says 'skip'. Should I just use that? Want to make
+      // the events readable and specific on the GA side.
+      eventLabel: 'Skip email'
+    });
+
+    this._close();
+  },
+
+  continue(e) {
+    e.preventDefault();
+
+    app.sendToGA('event', {
+      eventCategory: 'HomePage Interactions',
+      eventAction: 'button click',
+      eventLabel: 'On to the experiments'
+    });
+
+    this._close();
+  },
+
+  validate(email) {
+    return emailValidator(email);
```

Comment:
> `bob@bob` passed this validator.

@ckprice Yeah, email validation is hard. The validator library just copies Firefox's implementation of HTML5 email constraint validation, same approach as FxA. This means that some silly-looking emails will be counted as valid, like `root@localhost`. These are technically valid, according to the email RFC and the HTML5 validation spec.

I'm happy to explore other options, here or in a followup bug; let me know if you have a particular validation library in mind.